lib/shopping_cart.rb

```diff
@@ -0,0 +1,49 @@
+require './lib/product'
+
+class ShoppingCart
+  attr_reader :name, :capacity, :products, :total_number_of_products, :products_by_category
+
+  def initialize(name, capacity)
+    @name = name
+    @capacity = capacity.to_i
+    @products = []
+    @details = {name: @name, capacity: @capacity}
```

Comment:
although this is a simple line of code, we want to steer away from formatting in the initialize method, because if it becomes a habit, it can get a bit convoluted. So, I'd take this hash, and stick it into your details method at the bottom of this class.